pkg/cmd/cli/cmd/newbuild.go

```diff
@@ -0,0 +1,111 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/spf13/cobra"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	newcmd "github.com/openshift/origin/pkg/generate/app/cmd"
+)
+
+const (
+	newBuildLong = `Create a new build in OpenShift by specifying source code, and/or images.
+
+This command will try to create a BuildConfig for your application using images, or code that has
+a public repository located on your system. It will lookup the images on the
+local Docker installation (if available), a Docker registry, or an OpenShift image stream.
+If you specify a source code URL, it will set up a build that takes your source code and converts
+it into an image that can run inside of a pod. Local source must be in a git repository that has
+remote repository that the OpenShift instance can see.
+
+Once the BuildConfig will be created you may need to run a build with 'start-build'.`
+
+	newBuildExample = `  // Create a BuildConfig based on the source code in the current git repository (with a public remote) and a Docker image
+  $ %[1]s new-build . --docker-image=repo/langimage
+
+  // Create a NodeJS application based on the provided [image]~[source code] combination
+  $ %[1]s new-build openshift/nodejs-010-centos7~https://bitbucket.com/user/nodejs-app
+
+  // Create an application from a remote repository using its beta2 branch
+  $ %[1]s new-build https://github.com/openshift/ruby-hello-world#beta2`
+)
+
+// NewCmdNewBuild implements the OpenShift cli new-build command
+func NewCmdNewBuild(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	_, typer := f.Object()
+	config := newcmd.NewAppConfig(typer)
+
+	cmd := &cobra.Command{
+		Use:     "new-build (IMAGE | IMAGESTREAM | TEMPLATE | PATH | URL ...)",
```

Comment:
It's ok to have "IMAGE SOURCE" - the normal logic should associate them.

----- Original Message -----
> > +  $ %[1]s new-build . --docker-image=repo/langimage
> > +
> > +  // Create a NodeJS application based on the provided [image]~[source
> > code] combination
> > +  $ %[1]s new-build
> > openshift/nodejs-010-centos7~https://bitbucket.com/user/nodejs-app
> > +
> > +  // Create an application from a remote repository using its beta2 branch
> > +  $ %[1]s new-build https://github.com/openshift/ruby-hello-world#beta2`
> > +)
> > +
> > +// NewCmdNewBuild implements the OpenShift cli new-build command
> > +func NewCmdNewBuild(fullName string, f *clientcmd.Factory, out io.Writer)
> > *cobra.Command {
> > +	_, typer := f.Object()
> > +	config := newcmd.NewAppConfig(typer)
> > +
> > +	cmd := &cobra.Command{
> > +		Use:     "new-build (IMAGE | IMAGESTREAM | TEMPLATE | PATH | URL ...)",
> 
> IMAGE and IMAGESTREAM should be there only if associated with SOURCE
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2747/files#r31661607